doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+                        "type": "string"
+                    },
+                    "message": {
+                        "description": "Detailed information about the error.",
+                        "type": "string"
+                    },
+                    "loca": {
+                        "description": "Unique message token that can be used for localization in a frontend application.",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "status",
+                    "message",
+                    "loca"
+                ],
+                "additionalProperties": false
+            }
+        
+
+## Register [POST /register]
+
+This method can be used to register for the service. An user account will grant access to more functionality.
+
++ Request Register via form (application/x-www-form-urlencoded)
+    + Attributes
+        + username: the_username (string, required)
+        
+            The username must be between 3 and 20 signs long and may contain only letters a-z (lower- and upper-case), the numbers 0-9 and dots (.) as well as dashes (-).
+            
+        + password: somePassword (string, required) - A non-empty password
+        + email: an@email.com (string, required) - A valid email address
+        
++ Response 200 (application/json)
+    + Body
+    
+            {
+                "status":200,
+                "message":"Your account has been created successfully!",
+                "loca":"AUTH_REGISTER_USER_CREATED_SUCCESSFULLY"
+            }
+            
+    + Schema
+
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "error": {
+                        "description": "Textual description of HTTP status code",
+                        "type": "string"
+                    },
+                    "message": {
+                        "description": "Detailed information about the error.",
+                        "type": "string"
+                    },
+                    "loca": {
+                        "description": "Unique message token that can be used for localization in a frontend application.",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "status",
+                    "message",
+                    "loca"
+                ],
+                "additionalProperties": false
+            }
+        
++ Response 400 (application/json)
+    + Body
+
+            {
+                "status":400,
+                "error":"Bad Request",
+                "message":"You have to supply an username, a password and an email.",
```

Comment:
It's only about blueprint specification. It would be nice if I could list all possible error cases there, but adding a whole json response for each case is a bit much in my opinion (especially for the other API part later). Implementation is not a big deal.